kedro-org/kedro-viz
# Demo slicing pipeline final

## Description

This PR is for QA-ing purpose.

I've merged all the latest changes from 3 different PRs here (#1980 , #1996, #2003)


## QA notes
This PR should cover the following tasks

**_As a Kedro user, I want to be able to select multiple nodes within the flowchart so that I can slice the pipeline I am currently viewing_**
- [x] Notification on First Node Click: When I click on the first node, a notification should appear. This behavior should not occur when clicking on a pipeline (is it referred to as "pipeline" or "modular pipeline"?).
- [x]  Second Node Click with Shift: If I hold the Shift key and click on a second node, the metadata should close immediately, and the notification should disappear.
- [x]  when selecting a node from bottom to top, the order should be updated according to --from-node=bottom-node-id --to-node=top-node-id

**_As a Kedro user, I want to preview which nodes are included in my selection for slicing so that I can understand what my slice will be before creating a new slice_**

- [x] Chart Highlighting: The two nodes selected with Shift held should be highlighted with a darker blue background, and any nodes in between should be highlighted with a lighter blue background.
- [x]  Nodes List Highlighting: The nodes list on the left-hand side should also highlight all the nodes included in the newly selected pipeline.
- [x]  Call to Action: A button at the bottom should display the number of nodes selected and include a "Slice" button.
- [x]  Outside Click: Clicking outside the selection will un-select all the nodes, and go back to the default/previous page. This includes clicking on any other nodes outside the sliced pipeline

**_As a Kedro user, I want to view my sliced pipeline and its corresponding run command so that I can run this slice in the CLI for my Kedro project. I also want to be able to reset my sliced pipeline view so that I can go back to viewing the full Kedro pipeline_**

- [x]  Filtered Nodes View: A new view should display all the filtered nodes.
- [x]  Updated Node List: The node list should reflect all the newly filtered nodes.
- [x]  CLI Command Display: A command line at the bottom should show how to run the filtered nodes in the CLI.
- [x]  Close the sliced view: Click the exit button should take the user back to viewing the full kedro pipeline


## Checklist
